### ISSUE TITLE
Would be nice: "citations" attribute on nodes

### ISSUE THREAD

**Issue by liamhuber** [pyiron/pyiron_workflow#485]
Be able to know what to cite when you use a workflow.

To be defined at the class (i.e. decorator) level, so you can ask your workflow what you should cite. `Composite` nodes then just additionally compile over the set of citations among all children.

How to format these? A URL certainly (for websites), any other helper features for things like BibTex?

Should be easy to slap on top as an extra data field on `Node` and a loop-and-super on `Composite`.

**Comment by XzzX** [comment]
What about in general giving important objects a `metadata` field for such kind of information?

**Comment by liamhuber** [comment]
Yes, nesting the citation inside a more general metadata field sounds like a good idea. `Node` will still need to know about the explicit structure of the metadata (at least to the extent it knows where citations are) so that it can walk the graph and collect a set of all citations using something like `Node.get_citations -> set[str]`, but that doesn't preclude also having metadata the nodes aren't explicitly aware of.

**Comment by liamhuber** [comment]
Pre-`flowrep` issue
